src/classes/Database.php

```diff
@@ -190,4 +232,150 @@ public function validateUser(string $username, string $passwordHash): bool {
     public function getDbPath(): string {
         return $this->dbPath;
     }
+
+    private function migrationsTableExists(): bool {
+        if ($this->pdo === null) {
+            $this->connect();
+        }
+
+        if ($this->pdo === null) {
+            throw new \RuntimeException("Failed to connect to database");
+        }
+
+        $tableExists = $this->pdo->query("SELECT name FROM sqlite_master WHERE type='table' AND name='migrations'");
+        return $tableExists !== false && $tableExists->fetch() !== false;
+    }
+
+    public function getDatabaseVersion(): int {
+        Logger::debug("Getting database version");
+
+        if ($this->pdo === null) {
+            $this->connect();
+        }
+
+        if ($this->pdo === null) {
+            throw new \RuntimeException("Failed to connect to database");
+        }
+
+        try {
+            if (!$this->migrationsTableExists()) {
+                Logger::debug("Migrations table does not exist, returning version 0");
+                return 0;
+            }
+
+            $stmt = $this->pdo->query("SELECT MAX(version) FROM migrations");
+            if ($stmt === false) {
+                Logger::error("Failed to query migrations table");
+                return 0;
+            }
+            $version = $stmt->fetchColumn();
+
+            if ($version === false) {
+                Logger::debug("No migrations found, returning version 0");
+                return 0;
+            }
+
+            Logger::debug("Current database version", ['version' => $version]);
+            return (int) $version;
+        } catch (\PDOException $e) {
+            Logger::error("Error getting database version", ['error' => $e->getMessage()]);
+            throw $e;
+        }
+    }
+
+    public function setDatabaseVersion(int $version, string $name = 'Initial setup'): bool {
+        Logger::info("Setting database version", ['version' => $version, 'name' => $name]);
+
+        if ($this->pdo === null) {
+            $this->connect();
+        }
+
+        if ($this->pdo === null) {
+            throw new \RuntimeException("Failed to connect to database");
+        }
+
+        try {
+            // Check if this version already exists
+            $stmt = $this->pdo->prepare("SELECT COUNT(*) FROM migrations WHERE version = ?");
+            $stmt->execute([$version]);
+            $exists = (int) $stmt->fetchColumn() > 0;
+
+            if ($exists) {
+                Logger::debug("Version already exists, skipping", ['version' => $version]);
+                return true;
+            }
```

Comment:
@jetbrains-junie since there's an indey on the `version` column, this can be removed
```suggestion
 
```